compiler/rustc_lint/src/builtin.rs

```diff
@@ -997,18 +997,15 @@ impl<'tcx> LateLintPass<'tcx> for InvalidNoMangleItems {
                     self.check_no_mangle_on_generic_fn(cx, attr_span, it.owner_id.def_id);
                 }
             }
-            hir::ItemKind::Const(..) => {
+            hir::ItemKind::Const(ident, generics, ..) => {
                 if find_attr!(attrs, AttributeKind::NoMangle(..)) {
-                    // account for "pub const" (#45562)
-                    let start = cx
-                        .tcx
-                        .sess
-                        .source_map()
-                        .span_to_snippet(it.span)
-                        .map(|snippet| snippet.find("const").unwrap_or(0))
-                        .unwrap_or(0) as u32;
-                    // `const` is 5 chars
-                    let suggestion = it.span.with_hi(BytePos(it.span.lo().0 + start + 5));
+                    let suggestion =
+                        if generics.params.is_empty() && !generics.has_where_clause_predicates {
+                            // account for "pub const" (#45562)
+                            Some(it.vis_span.to(ident.span.shrink_to_lo()))
```

Comment:
My bad, I didn't look carefully enough. We actually want to include the span of the existing visibility if any since we're not suggesting `static` but `pub static` regardless of the original visibility which makes sense, of course.

```suggestion
                            Some(it.span.until(ident.span))
```

With `pub static ` (trailing space) as the replacement string. Again, take that with a pinch of salt, I'm yet again reviewing from my phone.

Thanks for impl'ing my optional suggestion. If this doesn't work out feel free to revert back to the preexisting code.